src/stdlib_io.fypp

```diff
@@ -111,7 +115,10 @@ contains
       !!     ...
       !!
       integer :: s
-      integer :: nrow, ncol, i
+      integer :: nrow, ncol, i, skiprows_, max_rows_
+
+      skiprows_ = max(optval(skiprows, 0), 0)
+      max_rows_ = optval(max_rows, -1)
```

Comment:
Should it be some checks to avoid `max_rows=0`? Or what is the expected behaviour in such a case?

Reply:
With `max_rows=0` you will get a 0-sized array. And I think that makes totally sense.